server/modlog.ts

```diff
@@ -137,8 +147,118 @@ export class Modlog {
 	sharedStreams: Map<ID, Streams.WriteStream | null> = new Map();
 	streams: Map<ModlogID, Streams.WriteStream | null> = new Map();
 
-	constructor(path: string) {
-		this.logPath = path;
+	readonly database: Database.Database;
+
+	readonly modlogInsertionQuery: Database.Statement<ModlogEntry>;
+	readonly FTSInsertionQuery: Database.Statement<[number, string?, string?, string?, string?]>;
+	readonly altsInsertionQuery: Database.Statement<[number, string]>;
+	readonly FTSAltsInsertionQuery: Database.Statement<[number, string]>;
+
+	readonly renameQuery: Database.Statement<[string, string]>;
+	readonly globalPunishmentsSearchQuery: Database.Statement<[string, string, string, number, ...string[]]>;
+	readonly insertionTransaction: Database.Transaction;
+	readonly altsInsertionTransaction: Database.Transaction;
+
+	constructor(flatFilePath: string, databasePath: string) {
+		this.logPath = flatFilePath;
+
+		this.database = new Database(databasePath);
+		this.database.exec("PRAGMA foreign_keys = ON;");
+
+		// Set up tables, etc
+		this.database.exec(FS(MODLOG_SCHEMA_PATH).readIfExistsSync());
+		if (Config.modlogftsextension) {
+			this.database.exec(`SELECT load_extension('native/fts_id_tokenizer.o')`);
+			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS modlog_fts USING fts5(note, userid, autoconfirmed_userid, action_taker_userid, content=modlog, content_rowid=modlog_id, tokenize='id_tokenizer')`);
+			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS alts_fts USING fts5(userid, content=alts, content_rowid=rowid, tokenize='id_tokenizer')`);
+		} else {
+			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS modlog_fts USING fts5(note, userid, autoconfirmed_userid, action_taker_userid, content=modlog, content_rowid=modlog_id)`);
+			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS alts_fts USING fts5(userid, content=alts, content_rowid=rowid)`);
```

Comment:
When you include the converter, each FTS table definitions have been duplicated 4 times, which is a pretty big red flag.

You can at least reduce the duplication here removing the branch, having 'tokenize' be either `'id_tokenizer'` or `'unicode61'` depending on whether the extension exists.

The JavaScript branch can be removed completely by having a `variables`/`settings` table which stores key/value pairs. Then `Config.modlogftsextension` could be moved from JS to the DB. We'll need something like this for storing the schema version anyway for migrations.

Reply:
Actually, it seems like `tokenize` can't take an SQL expression, which would require duplicating the table definitions if we moved it all to SQL. I'm not sure what the right thing to do here is. One idea would be to have something like `%TOKENIZER%` in the SQL file and `.replace` it when we load the script.

Reply:
`.replace()` seems reasonable